### ISSUE TITLE
Client side connection issues after upgrading Polkadot v0.9.23 to v0.9.26

### ISSUE THREAD

**Issue by Chralt98** [paritytech/substrate#12704]
```text
2022-11-14 16:54:27.169  INFO tokio-runtime-worker sub-libp2p: [Relaychain] 🔍 Discovered new external address for our node: /ip4/...  
2022-11-14 16:54:27.248  INFO tokio-runtime-worker sub-libp2p: [🌔 Zeitgeist Parachain] 🔍 Discovered new external address for our node: /ip4/...
```

The above is shown in the 0.9.23 version. This does not happen after the upgrade to 0.9.26. It's worth to mention, that this messages informs us, that there was a new external address found for **ip4**. After that message, the 0.9.23 version did found peers and sync with them.

We experience client side connection issues after the dependency update to polkadot-v0.9.26 in [this PR](https://github.com/zeitgeistpm/zeitgeist/pull/850). We would like to see, that the local node successfully synchronises with other nodes in the network. This correctly happens with the `polkadot-v0.9.23` version. I need to mention, that we used Moonbeam dependencies of Substrate and a couple of others before. Now we use the official dependencies. 

When I started the node for 0.9.23 (which does sync and found peers) and after that the 0.9.26 runtime, the updated version did sync too. But when the 0.9.26 run alone, it did not sync with the outer parachain nodes.

The following logs are noticeable with the 0.9.26 upgrade:
```bash
RUST_LOG=libp2p=trace ./target/debug/zeitgeist --chain=battery_station --tmp --detailed-log-output
```

```text
2022-11-14 17:16:08.423 TRACE tokio-runtime-worker sub-libp2p: [🌔 Zeitgeist Parachain] Ignoring self-reported non-global address /ip4/127.0.0.1/tcp/30334/ws from 12D3KooWEbhud9TrHYtzK2K1fpKeC4PjDu7bKnHTn8QYzyrfBJsK.    
2022-11-14 17:16:08.424 TRACE tokio-runtime-worker sub-libp2p: [🌔 Zeitgeist Parachain] Ignoring self-reported non-global address /ip4/192.168.1.3/tcp/30334/ws from 12D3KooWEbhud9TrHYtzK2K1fpKeC4PjDu7bKnHTn8QYzyrfBJsK.
...

2022-11-14 17:17:30.895 TRACE tokio-runtime-worker sub-libp2p: [Relaychain] Libp2p => Failed to reach PeerId("12D3KooWRuwKV1yt6fqPL6hsfPvJQ28pqmmTxNSDxEC4eLKagzjK"): An error occurred while negotiating the transport protocol(s) on a connection: [("/dns/bsr.zeitgeist.pm/tcp/30338/p2p/12D3KooWRuwKV1yt6fqPL6hsfPvJQ28pqmmTxNSDxEC4eLKagzjK", Other(Custom { kind: Other, error: Timeout })), ("/dns/bsr.zeitgeist.pm/tcp/30338/ws/p2p/12D3KooWRuwKV1yt6fqPL6hsfPvJQ28pqmmTxNSDxEC4eLKagzjK", Other(Custom { kind: Other, error: Timeout }))].    

...

2022-11-14 17:23:01.013 TRACE tokio-runtime-worker sub-libp2p: [🌔 Zeitgeist Parachain] Libp2p => Failed to reach PeerId("12D3KooWSBj4SXAz1ETTurW5PRAF1abyxXb9APAVs4vqBVr2NjRt"): An error occurred while negotiating the transport protocol(s) on a connection: [("/dns/bsr.zeitgeist.pm/tcp/30337/p2p/12D3KooWSBj4SXAz1ETTurW5PRAF1abyxXb9APAVs4vqBVr2NjRt", Other(Custom { kind: Other, error: Timeout })), ("/dns/bsr.zeitgeist.pm/tcp/30337/ws/p2p/12D3KooWSBj4SXAz1ETTurW5PRAF1abyxXb9APAVs4vqBVr2NjRt", Other(Custom { kind: Other, error: Timeout }))].
...

2022-11-14 17:22:55.965 TRACE tokio-runtime-worker sub-libp2p: [🌔 Zeitgeist Parachain] PSM => Connect(12D3KooWSBj4SXAz1ETTurW5PRAF1abyxXb9APAVs4vqBVr2NjRt, SetId(0)): Will start to connect at until Instant { t: 1910374941204 }  

...

2022-11-14 17:10:14.498 DEBUG tokio-runtime-worker libp2p_swarm: [🌔 Zeitgeist Parachain] Connection attempt to PeerId("12D3KooWSBj4SXAz1ETTurW5PRAF1abyxXb9APAVs4vqBVr2NjRt") failed with Transport([("/dns/bsr.zeitgeist.pm/tcp/30337/p2p/12D3KooWSBj4SXAz1ETTurW5PRAF1abyxXb9APAVs4vqBVr2NjRt", Other(Custom { kind: Other, error: Timeout })), ("/dns/bsr.zeitgeist.pm/tcp/30337/ws/p2p/12D3KooWSBj4SXAz1ETTurW5PRAF1abyxXb9APAVs4vqBVr2NjRt", Other(Custom { kind: Other, error: Timeout }))]).
```

**Comment by bkchr** [comment]
CC @altonen @dmitry-markin

**Comment by vivekvpandya** [comment]
I tried some debugging in libp2p and substrate networking and I find following noticeable difference in syncing peers (at 0.9.23) 
```
22-11-15 15:42:57.911 DEBUG tokio-runtime-worker trust_dns_proto::xfer::dns_handle: [🌔 Zeitgeist Parachain] querying: bootnode-1.zeitgeist.pm A
2022-11-15 15:42:57.911 TRACE tokio-runtime-worker netlink_sys::smol: [🌔 Zeitgeist Parachain] poll_recv_from: Ok((164, SocketAddr(nl_family=16, nl_pid=0, nl_groups=0))) 
```
and not syncing peers (at 0.9.26)
```
2022-11-15 14:58:00.234 DEBUG tokio-runtime-worker trust_dns_proto::xfer::dns_handle: [🌔 Zeitgeist Parachain] querying: bootnode-1.zeitgeist.pm A                                                                                                                                                            
 2022-11-15 14:58:00.234 TRACE tokio-runtime-worker netlink_sys::smol: [🌔 Zeitgeist Parachain] poll_recv_from: Err(Os { code: 11, kind: WouldBlock, message: "Resource temporarily unavailable" }) 
```

**Comment by vivekvpandya** [comment]
Also please consider https://substrate.stackexchange.com/questions/4990/parachain-not-syncing-with-peers

**Comment by Chralt98** [comment]
I investigated more time into it. I don't have the bootnode logs for you, but I have the following information.

Look at your test [here](https://github.com/paritytech/substrate/blob/1cca061ede75a4c693a3b71bd1b61d7275fe29e4/client/network/src/discovery.rs#L1138-L1191). On the 0.9.26 version we never get the case, that `Adding self-reported address ... to Kademlia DHT /battery_station_v2/kad` happens. So I assume, that the new 0.9.26 protocol is unsupported.

https://github.com/paritytech/substrate/blob/1cca061ede75a4c693a3b71bd1b61d7275fe29e4/client/network/src/discovery.rs#L316-L354

**Comment by vivekvpandya** [comment]
> I tried some debugging in libp2p and substrate networking and I find following noticeable difference in syncing peers (at 0.9.23)
> 
> ```
> 22-11-15 15:42:57.911 DEBUG tokio-runtime-worker trust_dns_proto::xfer::dns_handle: [🌔 Zeitgeist Parachain] querying: bootnode-1.zeitgeist.pm A
> 2022-11-15 15:42:57.911 TRACE tokio-runtime-worker netlink_sys::smol: [🌔 Zeitgeist Parachain] poll_recv_from: Ok((164, SocketAddr(nl_family=16, nl_pid=0, nl_groups=0))) 
> ```
> 
> and not syncing peers (at 0.9.26)
> 
> ```
> 2022-11-15 14:58:00.234 DEBUG tokio-runtime-worker trust_dns_proto::xfer::dns_handle: [🌔 Zeitgeist Parachain] querying: bootnode-1.zeitgeist.pm A                                                                                                                                                            
>  2022-11-15 14:58:00.234 TRACE tokio-runtime-worker netlink_sys::smol: [🌔 Zeitgeist Parachain] poll_recv_from: Err(Os { code: 11, kind: WouldBlock, message: "Resource temporarily unavailable" }) 
> ```

Please ignore this comment as it is not useful.

**Comment by vivekvpandya** [comment]
[fail_fine.txt](https://github.com/paritytech/substrate/files/10021680/fail_fine.txt)
[work_fine.txt](https://github.com/paritytech/substrate/files/10021681/work_fine.txt)

Here in fail_fine log I see following error
```
2022-11-16 17:02:44.909 DEBUG tokio-runtime-worker libp2p_swarm: [🌔 Zeitgeist Parachain] Connection attempt to PeerId("12D3KooWNpDxYCicWvNrR18QvkkAgVvBdsbh2woHfUGmKijj59aR") failed with Transport([("/dns/bootnode-0.zeitgeist.pm/tcp/30333/p2p/12D3KooWNpDxYCicWvNrR18QvkkAgVvBdsbh2woHfUGmKijj59aR", Other(Custom { kind: Other, error: Timeout })), ("/dns/bootnode-0.zeitgeist.pm/tcp/30333/ws/p2p/12D3KooWNpDxYCicWvNrR18QvkkAgVvBdsbh2woHfUGmKijj59aR", Other(Custom { kind: Other, error: Timeout }))]).    
2022-11-16 17:02:44.909 TRACE tokio-runtime-worker sub-libp2p: [🌔 Zeitgeist Parachain] Libp2p => Failed to reach PeerId("12D3KooWNpDxYCicWvNrR18QvkkAgVvBdsbh2woHfUGmKijj59aR"): An error occurred while negotiating the transport protocol(s) on a connection: [("/dns/bootnode-0.zeitgeist.pm/tcp/30333/p2p/12D3KooWNpDxYCicWvNrR18QvkkAgVvBdsbh2woHfUGmKijj59aR", Other(Custom { kind: Other, error: Timeout })), ("/dns/bootnode-0.zeitgeist.pm/tcp/30333/ws/p2p/12D3KooWNpDxYCicWvNrR18QvkkAgVvBdsbh2woHfUGmKijj59aR", Other(Custom { kind: Other, error: Timeout }))]. 
```

which I don't see in working case.

**Comment by Chralt98** [comment]
Yeah the protocol negotiation for the Zeitgeist protocols is not happening on the new version.

**Comment by Chralt98** [comment]
`INFO tokio-runtime-worker beefy: [Relaychain] 🥩 run BEEFY worker, best grandpa: #0.`

This above log happens on the old version, but not on the new one.

**Comment by altonen** [comment]
We did change some DHT-related stuff a while back (such as removing support for multiple DHTs) but I'm not convinced this issue is related to Kademlia. The nodes are discovered through some means and an attempt at establishing a connection is made to both but they timeout so I think the question here is why do they timeout.

If you don't have access to these bootnodes' logs, are you able to reproduce this issue by running the nodes locally and provides us with logs from both the "local bootnode" and local polkadot v0.9.26 node?

**Comment by Chralt98** [comment]
@altonen Thanks for your quick answer. I currently don't know how to start a local bootnode for v0.9.23 to simulate the behaviour to connect a v0.9.23 node with a new v0.9.26 node and then give you the log output of the v0.9.23 node. Can you please help me out there?

Apart from that I think [this](https://github.com/paritytech/polkadot/issues/5854) issue seems related to our problem. Maybe it's better to use a newer polkadot version indeed. (I also experience the log `WARN    async-std/runtime trust_dns_proto::xfer::dns_exchange: failed to associate send_message response to the sender`)

**Comment by altonen** [comment]
Yeah, the article you linked should allow you to setup a local network of nodes and test whether this issue is reproducible and if so, see what  happens on bootnode side from the logs.

That error message may be related to some response channel getting dropped due to request timeout and may be related to this error you've posted earlier.
```
... An error occurred while negotiating the transport protocol(s) on a connection: [("/dns/bootnode-0.zeitgeist.pm/tcp/30333/p2p/12D3KooWNpDxYCicWvNrR18QvkkAgVvBdsbh2woHfUGmKijj59aR", Other(Custom { kind: Other, error: Timeout })),
```

**Comment by Chralt98** [comment]
[debug_logs.zip](https://github.com/paritytech/substrate/files/10031515/debug_logs.zip)

Thanks @altonen , that you look into it.

**Comment by Chralt98** [comment]
After 1,5 hours the 0.9.26 node finally syncs with other 0.9.23 nodes on the network (even without running a local 0.9.23 node). Why couldn't we connect earlier?

**Comment by vivekvpandya** [comment]
Just to note this issue still continues with upgrade to 0.9.29 .

**Comment by vivekvpandya** [comment]
@bkchr I want to investigate on following
```
But the thing is for relay chain it shows ` [Relaychain] 🔍 Discovered new external address for our node: /ip4/3.xx.xx.53/tcp/7676/ws/p2p/12D3KooWLhkaoFSGV2JfXea1BGTEAv2R5PE4v2b1nHZMo1ZGtxxx ` And but for parachain it shows nothing like this
```
I see that in client/network/src/discovery.rs it never gets to execute inject_new_external_addr()
Can someone from substrate team help me if ip4 interface gets created or not for para-chain on local node?

**Comment by bkchr** [comment]
Cc @melekes @dmitry-markin can you help here please?

**Comment by melekes** [comment]
> Why could `listen_addrs` be empty for a peer?

I'd guess if `listen_addresses` in network config are empty. Otherwise, look for `Can't listen on XXX because` warning in logs.

**Comment by melekes** [comment]
> [debug_logs.zip](https://github.com/paritytech/substrate/files/10031515/debug_logs.zip)
> 
> Thanks @altonen , that you look into it.

In `debug_0.9.26_node.txt`:

```
2022-11-17 13:28:11.310  WARN main sub-libp2p: [Relaychain] Can't listen on /ip4/0.0.0.0/tcp/30334/ws because: Other(Custom { kind: Other, error: Other(A(A(A(
A(Transport(A(Transport(Os { code: 48, kind: AddrInUse, message: "Address already in use" })))))))) })
```

This means another process (instance of polkadot) is already listening on `30334`. Were you trying to spawn two processes on the same machine?

**Comment by vivekvpandya** [comment]
> 

I will get you a fresh logs, as Relaychain anyway works fine. So above error seems the case as you pointed, tried to spawn two nodes on same machine.

**Comment by vivekvpandya** [comment]
@melekes 
so I am getting following things (changed logs slightly to help myself
)
```
2023-01-18 13:02:46 [Relaychain] Listen Address: "/ip6/::/tcp/30334/ws"    
2023-01-18 13:02:46 [Relaychain] Listen Address: "/ip4/0.0.0.0/tcp/30334/ws"  
..
38:2023-01-18 12:56:04 [🌔 Zeitgeist Parachain] 🏷  Local node identity is: 12D3KooWRpKYJKRDhAXMMrqZWMRBHLUmn92dGppkAG1tfBW3oY6w    
39:2023-01-18 12:56:04 [🌔 Zeitgeist Parachain] Listen Address: "/ip6/::/tcp/30333"    
40:2023-01-18 12:56:04 [🌔 Zeitgeist Parachain] Listen Address: "/ip4/0.0.0.0/tcp/30333"   
```
However I don't see Can't Listen warning 
here is changed part of code 
```rust
     6         // Listen on multiaddresses.                                                                                                                                                                       
     5         for addr in &params.network_config.listen_addresses {                                                                                                                                              
+    4             info!(target: "sub-libp2p", "Listen Address: {:?}", addr);                                                                                                                                     
     3             if let Err(err) = Swarm::<Behaviour<B, Client>>::listen_on(&mut swarm, addr.clone()) {                                                                                                         ~    2                 info!(target: "sub-libp2p", "Can't listen on {} because: {:?}", addr, err)                                                                                                                 
     1             }                                                                                                                                                                                              
  435          }  
  ```

**Comment by melekes** [comment]
> However I don't see Can't Listen warning

That's good, right? So what are the issues you're experiencing?

**Comment by melekes** [comment]
> But the thing is for relay chain it shows ` [Relaychain] 🔍 Discovered new external address for our node: /ip4/3.xx.xx.53/tcp/7676/ws/p2p/12D3KooWLhkaoFSGV2JfXea1BGTEAv2R5PE4v2b1nHZMo1ZGtxxx ` And but for parachain it shows nothing like this

Could you please post debug logs of the parachain node here?

**Comment by Chralt98** [comment]
@bkchr https://github.com/libp2p/rust-libp2p/pull/2441#pullrequestreview-1011567446 your review comment is the cause of the problem here?

**Comment by vivekvpandya** [comment]
[log_parachain.txt](https://github.com/paritytech/substrate/files/10443561/log_parachain.txt)
@melekes let me know if you need more detailed log or any specific thing you were looking for is not included.

**Comment by melekes** [comment]
Do you happen to have logs of `/dns/bootnode-0.zeitgeist.pm/tcp/30333/p2p/12D3KooWNpDxYCicWvNrR18QvkkAgVvBdsbh2woHfUGmKijj59aR` or `/dns/bootnode-1.zeitgeist.pm/tcp/30333/ws/p2p/12D3KooWGLidEdRqggqc
ndA3Nsf6KcL3HbW7BdSGww6QNDkNnYV7`? Looks like we're failing to connect to both due to timeout.

**Comment by vivekvpandya** [comment]
> Do you happen to have logs of `/dns/bootnode-0.zeitgeist.pm/tcp/30333/p2p/12D3KooWNpDxYCicWvNrR18QvkkAgVvBdsbh2woHfUGmKijj59aR` or `/dns/bootnode-1.zeitgeist.pm/tcp/30333/ws/p2p/12D3KooWGLidEdRqggqc ndA3Nsf6KcL3HbW7BdSGww6QNDkNnYV7`? Looks like we're failing to connect to both due to timeout.
> 
> **To summarise the original issue: after upgrading from Polkadot v0.9.23 to v0.9.26 https://github.com/zeitgeistpm/zeitgeist parachain node no longer can connect to the two bootnodes due to timeout.**

I had get those logs once, but could not see anything which is concerning. Do you have specific thing that should be logged there?
then I can try restarting those nodes with certain flags to enable debugging and get logs.

**Comment by melekes** [comment]
> > Do you happen to have logs of `/dns/bootnode-0.zeitgeist.pm/tcp/30333/p2p/12D3KooWNpDxYCicWvNrR18QvkkAgVvBdsbh2woHfUGmKijj59aR` or `/dns/bootnode-1.zeitgeist.pm/tcp/30333/ws/p2p/12D3KooWGLidEdRqggqc ndA3Nsf6KcL3HbW7BdSGww6QNDkNnYV7`? Looks like we're failing to connect to both due to timeout.
> > **To summarise the original issue: after upgrading from Polkadot v0.9.23 to v0.9.26 https://github.com/zeitgeistpm/zeitgeist parachain node no longer can connect to the two bootnodes due to timeout.**
> 
> I had get those logs once, but could not see anything which is concerning. Do you have specific thing that should be logged there? then I can try restarting those nodes with certain flags to enable debugging and get logs.

Well, for starters, do they receive a connection request from `12D3KooWMj1VrY7319iTfy7qhKxdSz3HEWEC2bvuNyNYdAGEQGdt` (parachain node)? If yes, why it takes them longer than 20s to respond. If no, we'd have to go down to the networking layer (using [Wire](https://www.wireshark.org/)) and check that a) parachain node sends connection request b) bootnodes receive it.

**Comment by vivekvpandya** [comment]
@melekes I spawned two nodes on two different systems which are on same network via router. 
They have same code built from main branch, however they are not able to find each other as peer.
The command I used:
./target/debug/zeitgeist --dev --discover-local

@melekes also are you on Discord or any other IM service? Then we can work together faster.

**Comment by bkchr** [comment]
> @melekes I spawned two nodes on two different systems which are on same network via router.
> They have same code built from main branch, however they are not able to find each other as peer.
> The command I used:
> ./target/debug/zeitgeist --dev --discover-local

If the nodes running on two different systems, please specify one node as bootnode of the other node. We have mdns, but not as an ultra polished feature because you don't need this in real networks.

**Comment by vivekvpandya** [comment]
So I did a small experiment with local network where 192.168.29.90 has node running version v0.9.23 (which sync fine with bootnodes)
and other system 192.168.29.161 I run node with following two different commands 
1) working case
./target/debug/zeitgeist --bootnodes /ip4/192.168.29.90/tcp/3033/ws/p2p/12D3KooWNH2kJ6j61mVnpbVW1Tsyt7tNmaRqECaqbCKCPDhLYeNgg
[failing_para_log.txt](https://github.com/paritytech/substrate/files/10466090/failing_para_log.txt)
[working_para_log.txt](https://github.com/paritytech/substrate/files/10466091/working_para_log.txt)

2) not working case
[failing_para_log.txt](https://github.com/paritytech/substrate/files/10466093/failing_para_log.txt)
[working_para_log.txt](https://github.com/paritytech/substrate/files/10466095/working_para_log.txt)

./target/debug/zeitgeist --bootnodes /ip4/192.168.29.90/tcp/3033/p2p/12D3KooWNH2kJ6j61mVnpbVW1Tsyt7tNmaRqECaqbCKCPDhLYeNgg

attached logs for both.

**Comment by melekes** [comment]
> /ip4/192.168.29.90/tcp/3033/ws/p2p/12D3KooWNH2kJ6j61mVnpbVW1Tsyt7tNmaRqECaqbCKCPDhLYeNgg
> /ip4/192.168.29.90/tcp/3033/p2p/12D3KooWNH2kJ6j61mVnpbVW1Tsyt7tNmaRqECaqbCKCPDhLYeNgg

why did you add `/ws` protocol in the first case and did not in the second case?

**Comment by vivekvpandya** [comment]
> > /ip4/192.168.29.90/tcp/3033/ws/p2p/12D3KooWNH2kJ6j61mVnpbVW1Tsyt7tNmaRqECaqbCKCPDhLYeNgg
> > /ip4/192.168.29.90/tcp/3033/p2p/12D3KooWNH2kJ6j61mVnpbVW1Tsyt7tNmaRqECaqbCKCPDhLYeNgg
> 
> why did you add `/ws` protocol in the first case and did not in the second case?

That is how I can get a pass and failing case.
by default there is not /ws added, but when I add it manually it works like it used to work before.

**Comment by melekes** [comment]
> That is how I can get a pass and failing case.
> by default there is not /ws added, but when I add it manually it works like it used to work before.

So when you remove `/ws` from addr, it does not work w/ 0.9.26 (master) BUT works with 0.9.23
with `/ws` it works in both cases

correct?

**Comment by vivekvpandya** [comment]
> > That is how I can get a pass and failing case.
> > by default there is not /ws added, but when I add it manually it works like it used to work before.
> 
> So when you remove `/ws` from addr, it does not work w/ 0.9.26 (master) BUT works with 0.9.23 with `/ws` it works in both cases
> 
> correct?

Correct but I have never tried/ws with 0.9.23 as it works without it.

**Comment by melekes** [comment]
I've just tried to run `zeitgeist` with both `0.9.23` and `0.9.26` using the following cmd: `./target/debug/zeitgeist -l sub-libp2p=trace --bootnodes /dns/bootnode0.zeitgeist.pm/tcp/30333/p2p/12D3KooWNpDxYCicWvNrR18QvkkAgVvBdsbh2woHfUGmKijj59aR`

**In both cases I get timeout errors.**

**Comment by vivekvpandya** [comment]
> I've just tried to run `zeitgeist` with both `0.9.23` and `0.9.26` using the following cmd: `./target/debug/zeitgeist -l sub-libp2p=trace --bootnodes /dns/bootnode0.zeitgeist.pm/tcp/30333/p2p/12D3KooWNpDxYCicWvNrR18QvkkAgVvBdsbh2woHfUGmKijj59aR`
> 
> **In both cases I get timeout errors.**

Yes my experiment was on local network based setup. With dns based bootnodes even /ws tric does not get working case

**Comment by vivekvpandya** [comment]
On LAN a node at 192.168.29.90 runs zeitgiest node with polkadot version 0.9.23 (which last best working)
Now from other computer I start a node with  following commands
For not working case
./target/debug/zeitgeist --bootnodes /ip4/192.168.29.90/tcp/30333/p2p/12D3KooWNH2kJ6j61mVnpbVW1Tsyt7tNmaRqECaqbCKCPDhLYeNg -l sub-libp2p=trace

Logs are at for not working case: https://drive.google.com/file/d/1xdvskcLN1h9W4ixu27M58AnCPGQcDu7H/view?usp=sharing


./target/debug/zeitgeist --bootnodes /ip4/192.168.29.90/tcp/30333/ws/p2p/12D3KooWNH2kJ6j61mVnpbVW1Tsyt7tNmaRqECaqbCKCPDhLYeNg -l sub-libp2p=trace
	Logs for working case: https://drive.google.com/file/d/13KRhsejykotXAwjxtMrjqEv73ik-8sZm/view?usp=sharing 
	
The .zip file for logs also contain logs from node on 192.168.29.90

**Comment by dmitry-markin** [comment]
>     1. working case
>        ./target/debug/zeitgeist --bootnodes /ip4/192.168.29.90/tcp/3033/ws/p2p/12D3KooWNH2kJ6j61mVnpbVW1Tsyt7tNmaRqECaqbCKCPDhLYeNgg
>     2. not working case
>        ./target/debug/zeitgeist --bootnodes /ip4/192.168.29.90/tcp/3033/p2p/12D3KooWNH2kJ6j61mVnpbVW1Tsyt7tNmaRqECaqbCKCPDhLYeNgg

This is expected behavior, because the node listens on `/ws` endpoint by default, and on bare tcp only when run with `--dev` or `--validator`. See the command line option [help](https://github.com/paritytech/substrate/blob/master/client/cli/src/params/network_params.rs#L59-L65) and the [code](https://github.com/paritytech/substrate/blob/master/client/cli/src/params/network_params.rs#L165-L189).

**Comment by vivekvpandya** [comment]
> > ```
> > 1. working case
> >    ./target/debug/zeitgeist --bootnodes /ip4/192.168.29.90/tcp/3033/ws/p2p/12D3KooWNH2kJ6j61mVnpbVW1Tsyt7tNmaRqECaqbCKCPDhLYeNgg
> > 2. not working case
> >    ./target/debug/zeitgeist --bootnodes /ip4/192.168.29.90/tcp/3033/p2p/12D3KooWNH2kJ6j61mVnpbVW1Tsyt7tNmaRqECaqbCKCPDhLYeNgg
> > ```
> 
> This is expected behavior, because the node listens on `/ws` endpoint by default, and on bare tcp only when run with `--dev` or `--validator`. See the command line option [help](https://github.com/paritytech/substrate/blob/master/client/cli/src/params/network_params.rs#L59-L65) and the [code](https://github.com/paritytech/substrate/blob/master/client/cli/src/params/network_params.rs#L165-L189).

Agreed but with our public bootndes I am not able to connect with /ws , or using --dev and not using /ws in bootnode address.

**Comment by vivekvpandya** [comment]
I have captured packets for following commands
here 172.105.158.248 is external address for bootnode1-zeitgeist.pm 
 ./target/debug/zeitgeist --bootnodes /ip4/172.105.158.248/tcp/30333/ws/p2p/12D3KooWGLidEdRqggqcndA3Nsf6KcL3HbW7BdSGww6QNDkNnYV7

And a working case where I am running node on LAN address 192.168.29.90

![packet_capture](https://user-images.githubusercontent.com/5086212/215058101-9a5ac955-1e0f-4351-8907-e9a31b6e5377.png)

I see RST is being send to 172.105.158.248 and connection breaks and this goes in loop. However in working case I see RST is never sent to 192.168.29.90 and it successfully works.
Not sure reason behind RST, is local settings or something from code?
Here are wireshare packet_captures https://drive.google.com/file/d/1t3_-q0RHXeeLQyeE6-VQOCpcfTuNi_Vp/view?usp=sharing

**Comment by melekes** [comment]
So in the failing case we're terminating conn with a timeout (and send `RST`). In the working case we're successfully establishing the connection. Hmm 🤔

**Comment by vivekvpandya** [comment]
Reverse Experiment

on 192.168.29.90 I started node build with TOT main branch.

And from other machine 192.168.29.161 I start old working version (0.3.6) of zeitgeist node with following command:

`~/Downloads/zeitgeist_parachain --reserved-only --reserved-nodes /ip4/192.168.29.90/tcp/30333/ws/p2p/12D3KooWA9wshwxeNyvPdRN9Y2QnaPQNkprbQLGFbSTe3XypqJek`

I see both nodes find each other but they could not finalize any blocks.
For both nodes I see following log
`2023-01-30 16:32:32 [🔮 Zeitgeist Parachain] ⚙️  Syncing 139.9 bps, target=#325423 (1 peers), best: #45977 (0xb31f…3a66), finalized #0 (0x1bf2…e060), ⬇ 437.2kiB/s ⬆ 0.6kiB/s  `

**Comment by melekes** [comment]
@vivekvpandya are you sure your bootnode is reachable? for example I can't ping it

```
PING 172.105.158.248 (172.105.158.248): 56 data bytes                          │
Request timeout for icmp_seq 0                                                 │
Request timeout for icmp_seq 1                                                 │
Request timeout for icmp_seq 2                                                 │
Request timeout for icmp_seq 3                                                 │
Request timeout for icmp_seq 4                                                 │
Request timeout for icmp_seq 5
```

**Comment by melekes** [comment]
> I see both nodes find each other but they could not finalize any blocks.

are there any reasons you could see in the log as to why?
